src/main/java/duke/Storage/Storage.java

```diff
@@ -0,0 +1,127 @@
+ */
+public class Storage {
+    private final String path;
+
+    public Storage(String path) {
+        this.path = path;
+    }
+
+    /**
+     * Returns a list of Tasks that were stored in the tasks.txt file.
+     *
+     * @return a list of Tasks
+     * @throws DukeException If error writing to the files or no file detected.
+     */
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> list = new ArrayList<>();
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(path));
+            String line;
+            while ((line = br.readLine()) != null) {
+                if (!line.equals("Tasks:")) {
+                    Task task = convertToTask(line);
+                    list.add(task);
+                }
+            }
+            br.close();
+        } catch (IOException e1) {
+            File file = new File(path);
+            try {
+                file.createNewFile();
+            } catch (IOException e2) {
+                System.out.println("Error creating txt file");
+            }
+            throw new NoPreviousFileException("No previous file "
+                     + "detected, creating new one.");
+        }
+
+        return list;
+    }
+
+    private static Task convertToTask(String output) {
+        char type = output.charAt(1);
+        char done = output.charAt(4);
+        String taskAndDate = output.substring(7);
+
+        Task task;
+        if (type == 'T') {
+            task = new Todo(taskAndDate);
+        } else if (type == 'D') {
+            // Deadlines
+            if (taskAndDate.contains("(by: ")) {
+                String name = taskAndDate.split(" \\(by: ")[0];
+                String date = taskAndDate.split(" \\(by: ")[1];
+
+                // Chop of last ")"
+                date = date.substring(0, date.length() - 1);
+                task = new Deadline(name, " " + date);
+            } else {
+                task = new Deadline(taskAndDate, "");
+            }
+        } else {
+            // Events
+            if (taskAndDate.contains("(at: ")) {
+                String name = taskAndDate.split(" \\(at: ")[0];
+                String date = taskAndDate.split(" \\(at: ")[1];
+
+                // Chop of last ")"
+                date = date.substring(0, date.length() - 1);
+                task = new Event(name, " " + date);
+            } else {
+                task = new Event(taskAndDate, "");
+            }
+        }
+        task.setType(type);
+        if (done == 'X') {
+            task.setDone();
+        }
+        return task;
+    }
+
+    /**
+     * Updates the txt file to store and remember the tasks entered.
+     *
+     * @param tasklist This is taken in so that the txt file can be updated.
+     */
+    public void updateTxtFile(TaskList tasklist) {
+        BufferedWriter bw;
+        try {
+            bw = new BufferedWriter(new FileWriter(path));
+            for (Task t : tasklist.getTaskList()) {
```

Comment:
I noticed that in TaskList.java line 74, you used `Task t: taskList`. I think standardizing it to have whitespace around the colon is good.